analysis/task_analysis_surface/first_level_analysis_ss.py

```diff
@@ -0,0 +1,365 @@
+    if output: #checks if list is not empty
+        print(f'At least partial 1st level output exists for subject {sub}, session {ses}, run {run} and task {task}. This must be deleted for before generating new output.')
+        return    
+    
+    #run first level model 
+    labels, estimates, design_matrix, selected_confounds = run_first_level_model(sub,task,ses,run)         
+    
+    #define contrasts for task
+    contrasts = make_localizer_contrasts(design_matrix,task,selected_confounds)
+
+    for contrast in contrasts.keys():
+        print(contrast)
+        #calculate contrast maps for all contrasts
+        #F test to see if any nuisance regressor in contrast is significant
+        if contrast == 'nuisance_regressors':
+            contrast_output = compute_contrast(labels, estimates, contrasts[contrast],contrast_type='F')
+        #t test to see if specified individual contrast is significant
+        else:   
+            contrast_output = compute_contrast(labels, estimates, contrasts[contrast], contrast_type='t')
+        
+        #create dscalar files to be saved below
+        orig_cifti = glob.glob(f'../../../derivatives/ses-{ses}/sub-{sub}/ses-{ses}/func/sub-{sub}_ses-{ses}_task-{task}_rec-unco_run-{run}_space-fsLR_den-91k_bold.dtseries.nii')[0]
+        dscalar_output = create_dscalar(orig_cifti, contrast_output) 
+        
+        #create paths to output dir if not exist
+        derivatives_path = '../../../derivatives'
+        nilearn_output_path = os.path.join(derivatives_path, 'task_analysis_surface','first_level',f'sub-{sub}',f'ses-{ses}',f'task-{task}')
+        if not os.path.isdir(nilearn_output_path):
+            os.makedirs (nilearn_output_path)
+        
+        #save contrast maps to files
+        dscalar_output['effect_size'].to_filename(f'../../../derivatives/task_analysis_surface/first_level/sub-{sub}/ses-{ses}/task-{task}/sub-{sub}_ses-{ses}_task-{task}_rec-unco_run-{run}_contrast-{contrast}_effect_size.dscalar.nii')
+        dscalar_output['effect_variance'].to_filename(f'../../../derivatives/task_analysis_surface/first_level/sub-{sub}/ses-{ses}/task-{task}/sub-{sub}_ses-{ses}_task-{task}_rec-unco_run-{run}_contrast-{contrast}_effect_variance.dscalar.nii')
+        dscalar_output['z_score'].to_filename(f'../../../derivatives/task_analysis_surface/first_level/sub-{sub}/ses-{ses}/task-{task}/sub-{sub}_ses-{ses}_task-{task}_rec-unco_run-{run}_contrast-{contrast}_z_score.dscalar.nii')
+        dscalar_output['stat'].to_filename(f'../../../derivatives/task_analysis_surface/first_level/sub-{sub}/ses-{ses}/task-{task}/sub-{sub}_ses-{ses}_task-{task}_rec-unco_run-{run}_contrast-{contrast}_stat.dscalar.nii')
+        dscalar_output['p_value'].to_filename(f'../../../derivatives/task_analysis_surface/first_level/sub-{sub}/ses-{ses}/task-{task}/sub-{sub}_ses-{ses}_task-{task}_rec-unco_run-{run}_contrast-{contrast}_p_value.dscalar.nii')
```

Comment:
just to clarify, stat gets converted to pvalue and that gets converted to zscore by nilearn. so all three are effectively stats.

Reply:
yea thank you, that lab meeting discussion was really helpful! i was just maintaining the nilearn convention for naming